### PR TITLE
Build cleanup

### PR DESCRIPTION
**Pull request by Jake-Shadle**
* Moves the build shell script into build.rs
* Adds support (maybe?) for iOS

### CODE REVIEW

File: build.rs [hunk 1]
```diff
@@ -1,58 +1,160 @@
...
+    };
+
     println!("cargo:rustc-link-lib=framework=Metal");
     println!("cargo:rustc-link-lib=framework=AppKit");
     println!("cargo:rustc-link-lib=framework=QuartzCore");
     println!("cargo:rustc-link-lib=framework=IOKit");
     println!("cargo:rustc-link-lib=framework=IOSurface");
     println!("cargo:rustc-link-lib=dylib=c++");
-    println!("cargo:rustc-link-lib=static=MoltenVK");
+    println!("cargo:rustc-link-lib=static={}MoltenVK", target_name);
```
Comment (MaikKlein) [inline build.rs]: You rename the static lib to` lib(macos/ios)MoltenVK.a` right? So how does this work with external libs? The user would have to also rename it too?

Also why did you rename the lib? You only copy 1 lib afaik.

File: build.rs [hunk 2]
```diff
@@ -1,58 +1,160 @@
...
+
+                if counter >= 30 * 1000 {
+                    counter = 0;
+                    println!("still building MoltenVK");
+                }
+            }
+        });
+
+        let git_status = if Path::new("MoltenVK").exists() {
+            Command::new("git")
```
Comment (MaikKlein) [inline build.rs]: Hm shouldn't you call this from within Molten?

Comment (Jake-Shadle) [reply build.rs]: Ahh right, good catch.

File: build.rs (hunk 1 above)
Comment (Jake-Shadle) [reply build.rs]: This would mean that if you compiled for mac, and then for iOS, the second build would nuke the first, so everytime you switch your target platform it would have to redo a bunch of unnecessary work.
